cou/steps/backup.py

```diff
@@ -18,34 +18,40 @@
 import os
 from typing import Optional
 
-from cou.zaza_utils import model
-from cou.zaza_utils.upgrade_utils import (
-    extract_charm_name_from_url,
-    get_upgrade_candidates,
-)
+from juju.client._definitions import ApplicationStatus
 
+import cou.utils.juju_utils as utils
+from cou.exceptions import UnitNotFound
 
-def backup() -> str:
-    """Backup mysql database of openstack."""
-    logging.info("Backing up mysql database")
 
-    mysql_app = get_database_app()
-    mysql_leader = model.get_unit_from_name(model.get_lead_unit_name(mysql_app))
+async def backup(model_name: Optional[str] = None) -> str:
+    """Backup mysql database of openstack.
+
+    :param model_name: Optional model name.
+    :type args: Optional[str]
+    :return: Path of the local file from the backup.
+    :rtype: str
+    """
+    logging.info("Backing up mysql database")
+    mysql_app_config = await get_database_app(model_name)
+    if not mysql_app_config:
+        raise UnitNotFound()
```

Comment:
wouldn't it be better to raise this in the get_database_app function?

Reply:
done